cmd/terramate/cli/telemetry/telemetry.go

```diff
@@ -61,16 +84,56 @@ type Message struct {
 }
 
 // DetectPlatformFromEnv detects PlatformType based on environment variables.
-func DetectPlatformFromEnv() PlatformType {
+func DetectPlatformFromEnv(repo *git.Repository) (typ PlatformType, user string) {
 	if isEnvVarSet("GITHUB_ACTIONS") {
-		return PlatformGithub
+		typ = PlatformGithub
 	} else if isEnvVarSet("GITLAB_CI") {
-		return PlatformGitlab
+		typ = PlatformGitlab
+	} else if isEnvVarSet("BITBUCKET_BUILD_NUMBER") {
+		typ = PlatformBitBucket
+	} else if isEnvVarSet("TF_BUILD") {
+		typ = PlatformAzureDevops
+	} else if isEnvVarSet("APPVEYOR") {
+		typ = PlatformAppVeyor
+	} else if isEnvVarSet("bamboo.buildKey") {
+		typ = PlatformBamboo
+	} else if isEnvVarSet("BUDDY") {
+		typ = PlatformBuddyWorks
+	} else if isEnvVarSet("BUILDKITE") {
+		typ = PlatformBuildKite
+	} else if isEnvVarSet("CIRCLECI") {
+		typ = PlatformCircleCI
+	} else if isEnvVarSet("CIRRUS_CI") {
+		typ = PlatformCirrus
+	} else if isEnvVarSet("CODEBUILD_CI") {
+		typ = PlatformCodeBuild
+	} else if isEnvVarSet("HEROKU_TEST_RUN_ID") {
+		typ = PlatformHeroku
+	} else if strings.HasPrefix(os.Getenv("BUILD_TAG"), "hudson-") {
+		typ = PlatformHudson
+	} else if isEnvVarSet("JENKINS_URL") {
+		typ = PlatformJenkins
+	} else if os.Getenv("BuildRunner") == "MyGet" {
+		typ = PlatformMyGet
+	} else if isEnvVarSet("TEAMCITY_VERSION") {
+		typ = PlatformTeamCity
+	} else if isEnvVarSet("TRAVIS") {
+		typ = PlatformTravis
 	} else if isEnvVarSet("CI") {
-		return PlatformGenericCI
+		typ = PlatformGenericCI
 	} else {
-		return PlatformLocal
+		typ = PlatformLocal
+	}
```

Comment:
wdyt?

```go
	plats := map[string]PlatformType{
		"GITHUB_ACTIONS":         PlatformGithub,
		"GITLAB_CI":              PlatformGitlab,
		"BITBUCKET_BUILD_NUMBER": PlatformBitBucket,
		"TF_BUILD":               PlatformAzureDevops,
		"APPVEYOR":               PlatformAppVeyor,
		"bamboo.buildKey":        PlatformBamboo,
		"BUDDY":                  PlatformBuddyWorks,
		"BUILDKITE":              PlatformBuildKite,
		"CIRCLECI":               PlatformCircleCI,
		"CIRRUS_CI":              PlatformCirrus,
		"CODEBUILD_CI":           PlatformCodeBuild,
		"HEROKU_TEST_RUN_ID":     PlatformHeroku,
		"JENKINS_URL":            PlatformJenkins,
		"TEAMCITY_VERSION":       PlatformTeamCity,
		"TRAVIS":                 PlatformTravis,
		"CI":                     PlatformGenericCI,
	}

	typ = PlatformLocal
	for k, v := range plats {
		if isEnvVarSet(k) {
			typ = v
			break
		}
	}

	if typ == PlatformLocal {
		if strings.HasPrefix(os.Getenv("BUILD_TAG"), "hudson-") {
			typ = PlatformHudson
		} else if os.Getenv("BuildRunner") == "MyGet" {
			typ = PlatformMyGet
		}
	}
```

just opinions but I feel like these long *if* and *else* are hard to maintain.

Reply:
If it wasn't for the exceptions I would agree, but since they are there I find the switch block to be more obvious.

Reply:
the exceptions are in the `ifs` as well, they are not all handled the same, but all good